os-server/src/main/java/app/openschool/auth/AuthServiceImpl.java

```diff
@@ -83,9 +85,9 @@ public Optional<User> verifyAccount(String token) {
             .findVerificationTokenByToken(token)
             .orElseThrow(TokenValidationException::new);
 
-    checkTokenCondition(fetchedToken);
-
     User user = fetchedToken.getUser();
+    checkingConditionsBeforeVerification(fetchedToken, user);
```

Comment:
you still pass the user to the method, why retrieve the token and pass two arguments ?